### PR TITLE
Bugfix/cfo blob location issue

### PR DESCRIPTION
**Pull request by Chris-Lorch**
### Context
Minor changes to codebase to allow the running of 2021 and 2022 historic data, needed for the BFR run.

### Change proposed in this pull request
Minor changes do catch errors associated with the historic data.

### Guidance to review 
Ensure no overall change to logic. Code should now run with 2021 and 2022 data.

### Checklist
- [ ] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [ ] Your code builds clean without any errors or warnings
- [ ] You have run all unit/integration tests and they pass
- [ ] Your branch has been rebased onto main
- [ ] You have tested by running locally

### CODE REVIEW

File: data-pipeline/src/pipeline/pre_processing.py [hunk 1]
```diff
@@ -666,32 +665,40 @@ def build_maintained_school_data(
...
         maintained_schools_data_path,
-        encoding="utf8",
-        index_col=input_schemas.maintained_schools_master_list_index_col,
+        encoding="unicode-escape",
         usecols=input_schemas.maintained_schools_master_list.keys(),
-        dtype=input_schemas.maintained_schools_master_list,
     )
-
+    
+    mask = ((maintained_schools_list['Did Not Supply flag'] == '0').values
```
Comment (PsypherPunk) [inline data-pipeline/src/pipeline/pre_processing.py:672]: Could you run the linting commands over it?

```sh
poetry run black --config ./pyproject.toml src/ tests/
poetry run isort --settings-path ./pyproject.toml src/ tests/
```

Comment (Chris-Lorch) [reply data-pipeline/src/pipeline/pre_processing.py:672]: One day I will remember

Comment (PsypherPunk) [reply data-pipeline/src/pipeline/pre_processing.py:672]: I need to add a check to the pipeline…

**Review by PsypherPunk — APPROVED** [review]
Functionally, LGTM; I think some of the changes feed into the wider conversation around documentation (_what's_ being changed looks reasonable but _why_ those changes are needed isn't clear) but that's one to sort elsewhere, I think.

File: core-infrastructure/src/db/Core.Database/Properties/launchSettings.json
Comment (jrabbott) [inline core-infrastructure/src/db/Core.Database/Properties/launchSettings.json:1]: This file needs to be ignored, and not committed.

Comment (Chris-Lorch) [reply core-infrastructure/src/db/Core.Database/Properties/launchSettings.json:1]: added the path to the .gitignore file

**Review by PsypherPunk — APPROVED** [review]

File: data-pipeline/src/pipeline/pre_processing.py [hunk 2]
```diff
@@ -249,7 +246,12 @@ def prepare_ks4_data(ks4_path):
         inplace=True,
     )
 
-    return ks4[["AverageAttainment", "Progress8Measure", "Progress8Banding"]].dropna()
+    ks4 = (
+        ks4[["URN", "AverageAttainment", "Progress8Measure", "Progress8Banding"]]
+        .dropna()
+        .drop_duplicates()
```
Comment (colinbull) [inline data-pipeline/src/pipeline/pre_processing.py:252]: Just as an FYI you need to be really careful with `drop_duplicates` it doesn't always result in what your expecting. This caused a nasty bug in CDC I spotted at the weekend. In general prefer `result[~result.index.duplicated(keep='first')]`

Comment (colinbull) [reply data-pipeline/src/pipeline/pre_processing.py:252]: That being said if you are sure this has the right behaviour feel free to resolve this comment.

**Pull request merged**